scrapers/scrapers/spiders/sehgal_motors_spider.py

```diff
@@ -0,0 +1,99 @@
+import re
+import scrapy
+from scrapy.spiders import Rule
+from scrapy.linkextractors import LinkExtractor
+
+
+class SehgalMotorSpider(scrapy.Spider):
+    
+    name = 'sehgalmotors'
+    start_urls = ['https://www.sehgalmotors.pk/', ]
+    allowed_domain = 'sehgalmotors.pk'
+    link_extractor = LinkExtractor(allow='product/', allow_domains=allowed_domain)
+    rules = [Rule(link_extractor=link_extractor, callback='parse', follow=True)]
+    regex = re.compile(r'[\n\r\t ]+')
+
+    def parse(self, response):
+        try:
+            product = self.get_products(response)
+            product_sku = self.get_product_sku(product)
+            product_category = self.get_product_category(product)
+            product_name = self.get_product_name(product)
+            product_price = self.get_product_price(product)
+            product_images = self.get_product_images(product)
+            product_bread_crumbs = self.get_product_breadcrumbs(product)
+            product_info = self.get_product_info(product)
+            
+            yield{
+                'Product SKU': product_sku,
+                'Product Category': product_category,
+                'Product Name': product_name,
+                'Product Price': product_price,
+                'Product Images': product_images,
+                'Product Bread Crumbs': product_bread_crumbs,
+                'Product Usage Info': product_info
+            }
+            
+        except IndexError:
+            pass
+        
+        for link in self.link_extractor.extract_links(response):
+            yield scrapy.Request(link.url, callback=self.parse)
+
+
+    def get_products(self, response):
+        return response.css('body')
+    
+    
+    def get_product_category(self, product):
+        return product.css('nav.header_area a::text').getall()[1]
```

Comment:
Why we are indexing after extraction? Refactor your selector in that way that it should get exact value.